New term entry - d3 .scaleOrdinal() (#3993)
Codecademy/docs: content/d3/concepts/scaling/terms/domain/scaleOrdinal/scaleOrdinal.md

---
Title: 'scaleOrdinal()'
Description: 'Used to produce and obtain an ordinal scale'
Subjects:
  - 'Data visualization'
  - 'Web design'
Tags: 

>   - 'D3'

Reviewer: ```suggestion
  - 'D3'
  - 'Methods'
  - 'Objects'
```

> CatalogContent:

Reviewer: ```suggestion
CatalogContent:
  - 'learn-d3'
  - 'paths/data-science'
```

--- 

**scaleOrdinal()** function is used to generate and return an ordinal scale with the supplied domain and range. The array is set to be an empty array, if neither the range nor the domain are specified.

## Syntax

> It has two input parameters: domain and range:
> 
> scaleOrdinal(domain, range)
> 
> - The domain specifies the scale's maximum and minimum values.
> - Range. Values in the range corresponds to values in the domain.
> 
> d3.scaleOrdinal([[domain, ]range]);

Reviewer: ```suggestion
scaleOrdinal(domain, range)

It has two input parameters: domain and range:

- `domain`: This specifies the scale's maximum and minimum values.
- `range`: Values in the range correspond to values in the domain.
```

## Example

An array of date strings [1450, 1879, 1903, 1939] is connected to an array of word strings ["Printing press", "Electric light", "Airplane", "Computer"] by creating year scale using scaleOrdinal which maps the years to the inventions. 

```js
const year = d3.scaleOrdinal([1450, 1879, 1903, 1939], ["Printing press", "Electric light", "Airplane", "Computer"]);
```

```shell 
year(1450) // "Printing press"
year(1879) // "Electric light"
year(1903) // "Airplane"
year(1939) // "Computer"

> ```

Reviewer: what exactly is this? I don't see the above code printing anything.

Author: modified with different [example](https://github.com/Codecademy/docs/pull/3993/commits/8811f8e9090c721c248f3a8e6b984bfb23cfe95c), @dakshdeepHERE

CatalogContent:
  - 'learn-d3'
  - 'paths/data-science'
--- 

**scaleOrdinal()** function is used to generate and return an ordinal scale with the supplied domain and range. The array is set to be an empty array, if neither the range nor the domain are specified.

## Syntax

scaleOrdinal(domain, range)

It has two input parameters: domain and range:

- `domain`: This specifies the scale's maximum and minimum values.
- `range`: Values in the range correspond to values in the domain.

## Example

An array of date strings [1450, 1879, 1903, 1939] is connected to an array of word strings ["Printing press", "Electric light", "Airplane", "Computer"] by creating year scale using scaleOrdinal which maps the years to the inventions. 

```js
const inventors = ['Charles Babbage', 'Steve Jobs', 'Thomas Edison', 'Alexxander Graham']

const ordinalScale = d3.scaleOrdinal()
  .domain(inventors)
  .range(['The computer', 'iPhone', 'Lightbulb', 'Telephone'])
```

```shell 
ordinalScale('Charles Babbage'); // returns 'The computer';
ordinalScale('Steve Jobs'); // returns 'iPhone';
ordinalScale('Thomas Edison'); // returns 'Lightbulb';
ordinalScale('Alexander Graham'); // returns 'Telephone';

> ```

Reviewer: @A-Kiwams my main issue at start was this example that you've given. See the description that you have given for the snippet and then check the snippet. The description says you will map years with the invention. But the code you are trying to map the inventors with the inventions. Please fix this so we can move to the second review.

---
Title: 'scaleOrdinal()'

> Description: 'scaleOrdinal() function and examples'

Reviewer: make this description a small explanation of what scaleOrdinal() does and then explain it more below this metadata.

Author: Alright

Subjects:
  - 'Data visualization'
  - 'Web design'
Tags: 
  - 'D3'
CatalogContent:
--- 

**scaleOrdinal()** function is used to generate and return an ordinal scale with the supplied domain and range. The array is set to be an empty array, if neither the range nor the domain are specified.

## Syntax

It has two input parameters: domain and range:

scaleOrdinal(domain, range)

- The domain specifies the scale's maximum and minimum values.
- Range. Values in the range corresponds to values in the domain.

d3.scaleOrdinal([[domain, ]range]);

## Example

> Below example shows inventions corresponding to associated year of invention. 

Reviewer: This sentence should tell us what the `.scaleOrdinal()` is doing in the example below.

Author: Updated

```js
const year = d3.scaleOrdinal([1450, 1879, 1903, 1939], ["Printing press", "Electric light", "Airplane", "Computer"]);
```

> ## Output

Reviewer: remove this and add the shell block. Like you insert the code blocks in `pseudo` and `js` tags. Use the shell tag and then put the output of the code in it.

Author: Noted

If a value is given in the domain(input), results in a related value from range(output) value

```js
year(1450) // "Printing press"
year(1879) // "Electric light"
year(1903) // "Airplane"
year(1939) // "Computer"

> ```

Reviewer: what is this? The output for the above code?

Author: Output of the code

> ## D3 repository and docummentation
> [Repository](https://github.com/d3/d3-scale)
> [Docummentation](https://d3js.org/d3-scale/ordinal)
> 
> 
> 

Reviewer: ```suggestion
```

Author: Updated @dakshdeepHERE
